src/lib/analytics.ts

```diff
@@ -70,6 +70,19 @@ const safeAnalyticsTrack = (
   }
 }
 
+export function safeGetSegmentId(): string | null {
+  if (
+    typeof window !== undefined &&
+    window.analytics &&
+    window.analytics.user &&
+    typeof window.analytics.user === 'function'
```

Comment:
A cool TypeScript feature that might be helpful in this scenario, if you were unfamiliar

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

I'm 100% cool with this tho 👍

Reply:
Ya for whatever reason optional chaining doesn't work for this analytics def. Which is unintuitive, tell me if I'm wrong but when I tried to use optional chaining things still blew up. I think that's why they went with the heavy handed checks before with the tracking function.

Reply:
Oi okay ya 👍 trust that then.  Just was sharing in the event it was a feature you were unaware of!